frontend/src/scenes/App.tsx

```diff
@@ -27,48 +28,63 @@ const darkerScenes: Record<string, boolean> = {
     editFunnel: true,
     paths: true,
 }
+const plainScenes: Scene[] = [Scene.Ingestion, Scene.OrganizationCreateFirst, Scene.ProjectCreateFirst]
 
-const Toast = (): JSX.Element => {
+function Toast(): JSX.Element {
     return <ToastContainer autoClose={8000} transition={Slide} position="top-right" />
 }
 
 export const App = hot(_App)
-function _App(): JSX.Element {
+function _App(): JSX.Element | null {
     const { user } = useValues(userLogic)
-    const { currentTeam } = useValues(teamLogic)
+    const { currentOrganization, currentOrganizationLoading } = useValues(organizationLogic)
+    const { currentTeam, currentTeamLoading } = useValues(teamLogic)
     const { scene, params, loadedScenes } = useValues(sceneLogic)
     const { location } = useValues(router)
     const { replace } = useActions(router)
-    const [sidebarCollapsed, setSidebarCollapsed] = useState(typeof window !== 'undefined' && window.innerWidth <= 991) // used for legacy navigation [Sidebar.js]
+    // used for legacy navigation [Sidebar.js]
+    const [sidebarCollapsed, setSidebarCollapsed] = useState(typeof window !== 'undefined' && window.innerWidth <= 991)
     const { featureFlags } = useValues(featureFlagLogic)
 
     const Scene = loadedScenes[scene]?.component || (() => <SceneLoading />)
 
     useEffect(() => {
```

Comment:
This logic is quite heavy for the core <App /> component - can we extract it somewhere where it makes more semantic sense?

E.g. create a hook in `scenes/organization/hooks.ts` named `useCheckAuthentication`?

Reply:
I'd keep it in the same place at the moment, since it's already been here forever.